core/include/bitarithm.h

```diff
@@ -163,6 +163,47 @@ static inline unsigned bitarithm_lsb(unsigned v)
 }
 #endif
 
+/**
+ * @brief           Sets the masked bits in a word to the given value.
+ *
+ * @note            The mask should be compile-time constant so the compiler
+ *                  is able to optimize the entire loop away.
+ *
+ * @param       val The value to modify.
+ * @param[in]  mask The bitmask to apply.
+ * @param[in]  bits The new value of the masked bits.
+ *
+ */
+#define bitarithm_set_masked(val, mask, bits) do {                  \
+        const unsigned _mask = mask;                                \
```

Comment:
I would prefere a furthere indent for the code with exception of the closing `} while (0)` to see the structure of the macro a bit better.

Reply:
Might there be a problem for `uint32_t` parameters on platforms where `unsigned` is just 16 bit?

Reply:
Yes - for the get I could get away with a macro to end up at a generic version, but I can't return from a macro so I was hoping `unsigned` will be enough.

If we can use C11 `_Generic` in RIOT, maybe this will be the solution here - I don't think any of our tool chains is *that* old that it would not support it.

I'll see what I can figure out - enjoy your hike!

Reply:
IMO, if we're talking uint32_t and uint16_t, two implementations differing just in the type are preferable to a macro implementation. (That's also a coding convention - avoid macros if possible.)

Reply:
uint8_t was the original use-case, so that makes it 3.

Reply:
When ```bitarithm_ffs()``` is used, they'd be one line each, right?

Reply:
Shouldn't this be equivalent?

```
static inline void bitarithm_set_masked_u32(uint32_t *val, uint32_t mask, uint32_t bits)
{
    *(val) = ((*(val) & ~mask) | ((bits << (__builtin_ffs(mask) - 1)) & mask));
}
```

(generated code looks different though)

Reply:
Indeed, `__builtin_ffs` seems to produce exactly the same code on all our platforms. The long version `__builtin_ffsl(mask)` would have to be used for `uint32_t` since all our platforms define
```
#define __UINT32_TYPE__ long unsigned int
```